### ISSUE TITLE
[BUG] getLayeredPane() causes unexpected focus on InputComponent

### ISSUE THREAD

**Issue by jsfan3** [codenameone/CodenameOne#2710]
Please compare these two codes

**Code that reproduces a bug** _(the InputComponent get the focus without any reason, BUT on real devices WITHOUT the virtual keyboard opened)_:
```
Form hi = new Form("Test", new TextModeLayout(1,1));
        hi.add(new TextComponent().label("TestField"));
        Container layeredPane = hi.getLayeredPane();
        Log.p(layeredPane.getUIID());
        hi.show();
```
![bugreport_1](https://user-images.githubusercontent.com/1997316/52962478-a1444a00-339d-11e9-8c07-1ee28fa19504.png)

**Code that doesn't reproduce the bug** (the InputComponent doesn't get the focus until the user taps it)
```
Form hi = new Form("Test", new TextModeLayout(1,1));
        hi.add(new TextComponent().label("TestField"));
        hi.show();
        Container layeredPane = hi.getLayeredPane();
        Log.p(layeredPane.getUIID());
```
![bugreport_2](https://user-images.githubusercontent.com/1997316/52962599-f1bba780-339d-11e9-98ee-455bd010748d.png)

**Comment by codenameone** [comment]
Are you sure you posted the correct listings? I don't see any start edit or anything?

**Comment by jsfan3** [comment]
Yes, the codes are correct, please try them in the Simulator, so you will see the bug as in the screenshots.

**Comment by jsfan3** [comment]
To be more clear: the bug is that if `getLayeredPane()` is called BEFORE showing the Form, it causes that the first `InputComponet` gets the focus without any reason, without tapping and without `startEditing` (and also without opening the virtual keyboard). If `getLayeredPane()` is called AFTER showing the Form, there is no bug.
In the app I'm developing, I need to call `getLayeredPane()` before showing the Form.

**Comment by codenameone** [comment]
The bug is actually reversed. It seems the text field is added with an animation and as such the focus logic misses it because it isn't there yet when we decide on that. Getting the layer pane forces the flushing of the animation early enough to trigger the focus behavior.

This is something we should probably fix but it will also require a different fix that allows the UI to appear without the focus. 

Currently the workaround I could come up with is: 

````java
        Form hi = new Form("Test", new TextModeLayout(1,1));
        hi.getContentPane().setFocusable(true);
        hi.add(new TextComponent().label("TestField"));
        Container layeredPane = hi.getLayeredPane();
        Log.p(layeredPane.getUIID());
        hi.show();
        hi.getContentPane().setFocusable(false);
````

Because of the delicate nature of the actual fix I'll postpone it to the 7.0 cycle.

**Comment by jsfan3** [comment]
Thank you, I adapted your workaround to my code: it works :)

**Comment by liannacasper** [comment]
Is this still applicable?

**Comment by jsfan3** [comment]
@liannacasper Yes — I can still reproduce this on both the Codename One Simulator and a physical Android phone.

Retested on 5 September 2026 with Codename One **7.0.267**, using a fresh Maven project and the two original snippets, without the focus workaround:

- Simulator: Linux, Java 17, Galaxy S9 / Android skin.
- Device: Samsung Galaxy A20e (SM-A202F), Android 11 / API 30.

Both platforms give the same result:

| Position of `getLayeredPane()` | Focus immediately after showing the form | Native text editing / soft keyboard |
| --- | --- | --- |
| Before `hi.show()` | The `TextComponent`'s `TextField` receives focus automatically | Not started / not opened |
| After `hi.show()` | No component has focus | Not started / not opened |

I repeated each variant three times on each platform, checking at 0.25, 1 and 2.5 seconds after opening the form. The results were consistent, without touching the field or calling `requestFocus()` / `startEditingAsync()`. On Android, tapping the field in the second variant then starts editing and opens the keyboard normally.

The screenshots show the same before/after difference: the caret appears in the first variant even though the keyboard is closed. I have not retested iOS, but the issue is still applicable on the two platforms above.

Attachments: `android-before.png`, `android-after.png`, `simulator-before.png`, `simulator-after.png`, and `verification-results.txt`.

<img width="720" height="1560" alt="android-before.png" src="https://github.com/user-attachments/assets/d774dce0-7ea3-4bb0-a07f-9a5cb86d9a9e" />

<img width="720" height="1560" alt="android-after.png" src="https://github.com/user-attachments/assets/873f442a-e335-446c-ac8c-433421f7e0e8" />

<img width="475" height="1008" alt="simulator-before.png" src="https://github.com/user-attachments/assets/421024b3-5721-4fc3-8cc1-d07d5d6e5b4e" />

<img width="475" height="1008" alt="simulator-after.png" src="https://github.com/user-attachments/assets/bfeff6ad-4474-406b-9bc7-d9555a51e4ce" />

[verification-results.txt](https://github.com/user-attachments/files/31855600/verification-results.txt)
